ga4gh/datamodel/biodata.py

```diff
@@ -0,0 +1,226 @@
+"""
+Biodata objects
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import datetime
+
+import ga4gh.datamodel as datamodel
+import ga4gh.protocol as protocol
+import ga4gh.exceptions as exceptions
+
+
+class AbstractBioSample(datamodel.DatamodelObject):
+    """
+    This class represents an abstract BioSample object.
+    It sets default values and getters, as well as the
+    toProtocolElement function.
+    """
+    compoundIdClass = datamodel.BioSampleCompoundId
+
+    def __init__(self, parentContainer, localId, individualName=None):
+        super(AbstractBioSample, self).__init__(parentContainer, localId)
+        self._createDateTime = datetime.datetime.now().isoformat()
+        self._updateDateTime = datetime.datetime.now().isoformat()
+        self._description = ""
+        self._disease = None
+        self._info = {}
+        self._name = localId
+        self._individualName = individualName
+        self._individualId = None
+
+    def toProtocolElement(self):
+        bioSample = protocol.BioSample()
+        bioSample.createDateTime = self.getCreateDateTime()
+        bioSample.description = self.getDescription()
+        bioSample.disease = self.getDisease()
+        bioSample.id = self.getId()
+        bioSample.individualId = self.getIndividualId()
+        bioSample.info = self.getInfo()
+        bioSample.name = self.getName()
+        bioSample.updateDateTime = self.getUpdateDateTime()
+        return bioSample
+
+    def getIndividualId(self):
+        """
+        Return a string identifier using the datasetId
+        and localId. LocalIds of individuals are expected
+        to have the same localId as a BioSample.
+        :return:
+        """
+        datasetId = self.getParentContainer().getCompoundId()
+        compoundId = datamodel.IndividualCompoundId(
+            datasetId, self._individualName)
+        return str(compoundId)
+
+    def getCreateDateTime(self):
+        return self._createDateTime
+
+    def getUpdateDateTime(self):
+        return self._updateDateTime
+
+    def getDescription(self):
+        return self._description
+
+    def getDisease(self):
+        return self._disease
+
+    def getInfo(self):
+        return self._info
+
+    def getName(self):
+        return self._name
+
+
+class JsonBioSample(AbstractBioSample, datamodel.MetadataSidecarMixin):
+    """
+    Allows a BioSample object to be populated using JSON
+    metadata information by passing a filepath during
+    instantiation.
+    b = JsonBioSample(dataset, name, filepath)
+    """
+    def __init__(self, parentContainer, localId, filePath):
+        super(JsonBioSample, self).__init__(parentContainer, localId)
+        self._filePath = filePath
+        self.loadSidecar(filePath)
+        self.validate()
```

Comment:
I think the mixin is overcomplicating things a bit. How about this
```python
metadata = readJsonMetadata(filePath)
self._description = metadata.get("description", self._description)
# And so on for the rest of the attributes.
```

Then, this class is nice and simple and doesn't need all the overrides.